analyzers/src/SonarAnalyzer.CSharp/Rules/UnchangedLocalVariablesShouldBeConst.cs

```diff
@@ -147,44 +139,27 @@ protected override void Initialize(SonarAnalysisContext context)
                 .Any(IsMutatingUse);
 
             static bool IsMethodLike(SyntaxNode arg) =>
-                arg is BaseMethodDeclarationSyntax ||
-                arg is IndexerDeclarationSyntax ||
-                arg is AccessorDeclarationSyntax ||
-                arg is LambdaExpressionSyntax;
+                arg is BaseMethodDeclarationSyntax
+                || arg is IndexerDeclarationSyntax
+                || arg is AccessorDeclarationSyntax
+                || arg is LambdaExpressionSyntax;
 
-            bool MatchesIdentifier(IdentifierNameSyntax id)
-            {
-                var symbol = semanticModel.GetSymbolInfo(id).Symbol;
-                return Equals(parameterSymbol, symbol);
-            }
+            bool MatchesIdentifier(IdentifierNameSyntax id) =>
+                Equals(parameterSymbol, semanticModel.GetSymbolInfo(id).Symbol);
         }
 
-        private static bool IsMutatingUse(IdentifierNameSyntax id)
-        {
-            var parent = id.Parent;
-            if (parent is AssignmentExpressionSyntax assignmentExpression &&
-                Equals(id, assignmentExpression?.Left))
-            {
-                return true;
-            }
-
-            if (parent is PostfixUnaryExpressionSyntax ||
-                parent is PrefixUnaryExpressionSyntax)
+        private static bool IsMutatingUse(IdentifierNameSyntax identifier) =>
+            identifier.Parent switch
```

Comment:
Like a new RSPEC? That would be a valuable one. Because it took me some time to analyze the flow and understand that it can be actually simplified like this.

Reply:
#3813